types/ember__component/template-only.d.ts

```diff
@@ -0,0 +1,5 @@
+export class TemplateOnlyComponent {
```

Comment:
We should export this as an interface instead of as a class, so that people do not think they can construct it.

```suggestion
export interface TemplateOnlyComponent {
```

Reply:
An alternative approach that might actually be even better, and which I've been thinking about as a pattern for this kind of publicly-consumable-interface-but-privately-constructable-type:

```ts
class _TemplateOnlyComponent {
  declare private brand: 'TemplateOnlyComponent';
  toString(): string;
}

export interface TemplateOnlyComponent extends _TemplateOnlyComponent {}

export default function templateOnly(moduleName?: string): TemplateOnlyComponent;
```

This creates a "nominal"-ish type, which is useful here since `{ toString(): string }` is, well, nearly everything. 😂

Reply:
…note that if you do that, please definitely comment on it so that it's clear *why* we're doing it.

Reply:
Yeah, I wasn't sure what to do here. In the first version I did, I defined, but didn't export it.

Reply:
Thinking you could just use `ReturnType` if you needed it.

Reply:
Yeah, I think being able to name the type makes sense, but we *have* historically chosen not to do that, since Ember itself doesn’t have a defined approach for these things. Another thing to tackle in upcoming RFCs. I’ll leave it to your discretion here which way to do it, but we definitely want the nominalish class I think.